packages/matter.js/src/crypto/Key.ts

```diff
@@ -518,21 +523,21 @@ export function Key(properties: Partial<Key>) {
         const crv = that.crv;
         switch (crv) {
             case CurveType.p256:
-            case CurveType.p384:
-            case CurveType.p521:
+                // We can add the other point types easily by exposing more from @noble/curves
+                // case CurveType.p384:
+                // case CurveType.p521:
                 break;
 
             default:
                 throw new KeyError(`Unsupported elliptic curve ${crv}`);
         }
 
         // Compute
-        const elliptic = new ec(crv.toLowerCase().replace(/-/g, ""));
-        const ecKey = elliptic.keyFromPrivate(that.privateKey).getPublic();
+        const ecKey = ProjectivePoint.fromPrivateKey(that.privateKey);
 
         // Install
-        that.xBits = new ByteArray(ecKey.getX().toArray());
-        that.yBits = new ByteArray(ecKey.getY().toArray());
+        that.xBits = numberToBytesBE(ecKey.x, 32); // ???
```

Comment:
I think this is correct for p256, I'd probably create a variable and set it in the `crv` switch statement above.  Might be a field on `ecKey` you could use for this, e.g. using "k" from the hasher then `(2 * k / 128)`.  But not clear to me k is accessible just from reading the source.

Reply:
No, not accessible. I added a normal variable for now